### PR TITLE
Dedupe content

### PR DESCRIPTION
**Pull request by ralphiee22**
### Summary

Remove duplicates from popular and resume api responses. 
Also only filter against content nodes that currently exist in the database.

### Reviewer guidance

Changes are on http://kolibri-beta.learningequality.org/learn/#/recommended.
sign in as admin to see resume response.
Pay attention to load time?

### References

#3861 

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by DXCanas** [comment]
There are some things happening with recommended page that are making me wonder if this is the best approach. @jtamiace 

Because of #3847, we're considering removing the "recommended content" page and treating the items that pop up (and their duplicates) in the carousel like we do in search results:
search results page

Not entirely sure what's going on here, but would this remove items duplicate content items in different topic trees?

**Comment by ralphiee22** [comment]
That is a correct assumption and actually quite problematic. For recommended we basically pick a random content node which could be in any channel and return that to the user. 
For resume, its more of an issue I am worried about because we could return content nodes for a channel which they haven't even explored.

File: kolibri/content/api.py [hunk 1]
```diff
@@ -217,22 +217,34 @@ def filter_popular(self, queryset, name, value):
...
+
+        # remove duplicate content items
+        deduped_list = []
+        content_ids = set()
+        for node in most_popular:
+            if node.content_id not in content_ids:
+                deduped_list.append(node)
+                content_ids.add(node.content_id)
+
+        queryset = most_popular.filter(id__in=[node.id for node in deduped_list])
```
Comment (indirectlylit) [inline kolibri/content/api.py:237]: is it necessary to do a second query? can you not just return the `deduped_list`?

Comment (ralphiee22) [reply kolibri/content/api.py:237]: the result of this query is returned to another filter method (ex. `filter_by_role`). And the logic in there expects another queryset. If we return a list then we can't do methods on it like `.exclude`.

**Review by indirectlylit — APPROVED** [review]

**Pull request merged**